Octokit/Models/Response/StringEnum.cs

```diff
@@ -66,7 +68,7 @@ public bool TryParse(out TEnum value)
                 return true;
             }
 
-            if (string.IsNullOrEmpty(StringValue))
+            if (StringValue == null)
```

Comment:
How can `StringValue` be null at this point?

Reply:
It seems to be `null` in the case where you have a class member of this type that is unitialized.

Reply:
I added a test to repro this here: https://github.com/octokit/octokit.net/pull/1760/files#diff-728143ef4b5a14ceed6d74d06f9c9df7R62

this is certainly doing my head in!

Reply:
Couldn't this potentially lead to incorrect behavior?

If you have an uninitialized property, it means that the payload is missing the property, which means it's optional, but Octokit has modeled it as required (by it not being nullable).

By just silently falling back to the default value, you could potentially give the user the incorrect default (who decides what the "default value" is?)

Reply:
right... I was losing sight of the fact we only use this `StringEnum<T>` in response models, because the unit tests are using dummy classes where it's possible to declare classes with uninitialised members and so on.

But from a technical point of view - the same would be true of an `enum` property on a response class now wouldn't it?  If it was not marked nullable, and the API response didnt specify it, then we will be providing the "incorrect" default enum value on the response model instead of a null.  

I guess the question is, should an incorrectly not nullable `StringEnum<T>` behave the same as an incorrectly not nullable regular `enum`?

Reply:
> But from a technical point of view - the same would be true of an enum property on a response class now wouldn't it?

Yes, but that doesn't make it any less wrong 😉 There's a reason you should always declare all value-type properties as nullable when binding to external input. It's the only way to check whether that input actually existed, or if it's just the default (uninitialized) value.

> I guess the question is, should an incorrectly not nullable StringEnum<T> behave the same as an incorrectly not nullable regular enum?

I vote "no", simply because it's a nice way to "flush out" mistakes where we've assumed something was required, but was actually optional.

Reply:
Let's say you have a model:

Reply:
Let's say you have a model:

```csharp
public class Account
{
    public Guid Id { get; set; }
    public decimal Amount { get; set; }
}
```

And someone posts

```json
{ "id": "3d216536-0869-4c9d-b2b7-2a735498c634" }